src/android/CodePush.java

```diff
@@ -109,17 +113,17 @@ private boolean execUpdateSuccess(CallbackContext callbackContext) {
             /* save reporting status for first install */
             try {
                 String appVersion = Utilities.getAppVersionName(cordova.getActivity());
-                codePushReportingManager.reportStatus(CodePushReportingManager.Status.STORE_VERSION, null, appVersion, mainWebView.getPreferences().getString(DEPLOYMENT_KEY_PREFERENCE, null), this.mainWebView);
+                codePushReportingManager.reportStatus(new StatusReport(ReportingStatus.STORE_VERSION, null, appVersion, mainWebView.getPreferences().getString(DEPLOYMENT_KEY_PREFERENCE, null)), this.mainWebView);
```

Comment:
i think this method should be renamed from `execUpdateSuccess` to something more telling of what it actually does. as far as i can tell, it's more of a `execConfirmInstall`. the issue is that  `execUpdateSuccess` implies that an update has just been installed and succeeded, but that is not always the case--if i'm correct, it's called every time notifyApplicationReady happens, which means any number of situations could be the case, most of which are actually not the result of installing updates

additionally i think there needs to be a comment in here to explain this if/else chain behavior on a more general level
```
// Report the current installation to metrics.
// If it is the first run of a store version app, an installation that needs confirmation, or an installation for which the previous report failed, we need to report to metrics.
// If it is not any of those cases, the installation has already been reported to metrics so we don't need to do so again.
```